client/sttp-client/src/main/scala/sttp/tapir/client/sttp/EndpointToSttpClient.scala

```diff
@@ -185,7 +185,8 @@ private[sttp] class EndpointToSttpClient[R](clientOptions: SttpClientOptions, ws
         req.multipartBody(parts.toList)
     }
 
-    req2.contentType(codec.format.mediaType)
+    if (req2.contentType.isEmpty) req2.contentType(codec.format.mediaType)
+    else req2
```

Comment:
Actually, it seems that content-type header is always set thanks to `req.body` methods (which in turn call `req.withBody`). Except for the case of `req.multipartBody` wich only calls `req.copy`.

I'm not really sure to understand why `req2.contentType` was called in every case then:
- Just to make sure content-type header was always set?
- `req.body` methods were not setting the header properly at the time this was written?
- The `codec.format.mediaType` value may be different from `body.defaultContentType` ?